### PR TITLE
Bug/421 password validation UI

### PR DESCRIPTION
**Pull request by HaseebUllahAbbasi**
Now user can see what is password status how strong is it as it will be approved or not

### CODE REVIEW

**Comment by HaseebUllahAbbasi** [comment]
but there is overlapping issue is still there I will figure it out, and that is not only on one screen then it has to implmented on call the screens when the keyboard opens

File: src/containers/CreateAccountPasswordContainer.tsx [hunk 1]
```diff
@@ -18,19 +18,37 @@ import { Props } from '../screens/CreateAccountPasswordScreen';
...
     const [showUsernameErrorModal, setShowUsernameErrorModal] = useState(false);
     const [error, setError] = useState(false);
     const user = useUserStore().user;
     const errorStore = useErrorStore();
-
+    const passwordRegex = /^(?=.*\d)(?=.*[A-Z])(?=.*[a-z])(?=.*[^\w\d\s:])([^\s]){12,}$/;
     useEffect(() => {
-        setError(!(password === password2));
+        if (password.length > 0) {
+            const isValid = passwordRegex.test(password);
```
Comment (theblockstalk) [inline src/containers/CreateAccountPasswordContainer.tsx:36]: this is already done within `user.savePassword` function and does not need to be done twice. Correct error handling of `user.savePassword` can handle the condition you have here already.

Comment (theblockstalk) [reply src/containers/CreateAccountPasswordContainer.tsx:36]: please use `user.savePassword` here instead.

File: src/containers/CreateAccountPasswordContainer.tsx [hunk 2]
```diff
@@ -18,19 +18,37 @@ import { Props } from '../screens/CreateAccountPasswordScreen';
...
     const [loading, setLoading] = useState(false);
     const [trxUrl, setTrxUrl] = useState('');
     const [showModal, setShowModal] = useState(false);
     const [showUsernameErrorModal, setShowUsernameErrorModal] = useState(false);
     const [error, setError] = useState(false);
     const user = useUserStore().user;
     const errorStore = useErrorStore();
-
+    const passwordRegex = /^(?=.*\d)(?=.*[A-Z])(?=.*[a-z])(?=.*[^\w\d\s:])([^\s]){12,}$/;
     useEffect(() => {
```
Comment (theblockstalk) [inline src/containers/CreateAccountPasswordContainer.tsx:34]: make sure all functions have a line space before and after

File: src/containers/CreateAccountPasswordContainer.tsx [hunk 3]
```diff
@@ -96,23 +114,31 @@ export default function CreateAccountPasswordContainer({ navigation }: Props) {
...
+                                    style={errorMessage.length > 0 ? errorStyles.labelError : styles.labelText}
+                                >
                                     Password
                                 </TP>
                                 <TPasswordInput
                                     value={password}
                                     onChangeText={setPassword}
                                     errorText={errorMessage}
-                                    outlineColor={error ? theme.colors.error : theme.colors.primary}
+                                    outlineColor={errorMessage.length > 0 ? theme.colors.error : theme.colors.primary}
```
Comment (theblockstalk) [inline src/containers/CreateAccountPasswordContainer.tsx:127]: better to use the `error` state as it is used for both the modal + validation errors

File: src/containers/CreateAccountPasswordContainer.tsx [hunk 4]
```diff
@@ -96,23 +114,31 @@ export default function CreateAccountPasswordContainer({ navigation }: Props) {
...
                                     style={commonStyles.marginBottom}
                                 />
                             </View>
                             <View>
-                                <TP style={error ? errorStyles.labelError : styles.labelText}>Confirm Password</TP>
+                                <TP style={confirmErrorMessage.length > 0 ? errorStyles.labelError : styles.labelText}>
```
Comment (theblockstalk) [inline src/containers/CreateAccountPasswordContainer.tsx:132]: same as L127

File: src/containers/CreateAccountPasswordContainer.tsx (hunk 2 above)
Comment (HaseebUllahAbbasi) [reply src/containers/CreateAccountPasswordContainer.tsx:34]: removed in my new commit

File: src/containers/CreateAccountPasswordContainer.tsx (hunk 3 above)
Comment (HaseebUllahAbbasi) [reply src/containers/CreateAccountPasswordContainer.tsx:127]: I used it as there are two fields one for password and one for confirm password so this one is for password

File: src/containers/CreateAccountPasswordContainer.tsx (hunk 4 above)
Comment (HaseebUllahAbbasi) [reply src/containers/CreateAccountPasswordContainer.tsx:132]: I used it as there are two fields one for password and one for confirm password so this one is for confirm password, it is true when user confirm password doesn't matches the password

Comment (HaseebUllahAbbasi) [reply src/containers/CreateAccountPasswordContainer.tsx:132]: @Rebal17 @theblockstalk

**Pull request merged**